.github/workflows/swift-toolchain.yml

```diff
@@ -3896,11 +3904,38 @@ jobs:
           name: ${{ matrix.os }}-${{ matrix.arch }}-debugging_tools
           path: ${{ github.workspace }}/BuildRoot/Library
 
+  # Determines whether we're using trusted signing in this repository. If so and we're signing, downloads the trusted signing DLL to reduce boilerplate in steps that use it.
+  # Setting an output is necessary because the `env` and `secrets` contexts are not always available in `if` clauses.
+  configure_signing:
+    if: inputs.build_os == 'Windows'
+    name: Configure signing
+    runs-on: ${{ inputs.default_build_runner }}
+    outputs:
+      uses_trusted_signing: ${{ steps.test_trusted_signing.outputs.uses_trusted_signing }}
+
+    steps:
+      - name: Test whether we're using trusted signing
+        id: test_trusted_signing
+        run: |
+          "uses_trusted_signing=${{ secrets.TRUSTED_SIGNING_ACCOUNT != ''}}" | Out-File -FilePath ${env:GITHUB_OUTPUT} -Encoding utf8 -Append
+      - uses: nuget/setup-nuget@v2
+        if: inputs.signed && steps.test_trusted_signing.outputs.uses_trusted_signing == 'true'
+      - run: |
+          nuget install -x -OutputDirectory ${{ github.workspace }} Microsoft.Trusted.Signing.Client
```

Comment:
How long does this step take? Is it worth it to not re-run it on every jon?

Reply:
I am also concerned about Microsoft changing the layout at some point so all of the necessary resource files aren't in this directory but that's probably good enough for now. `nuget` has been flaky in the past too so reducing it to a single command is not a bad idea, we can also fail early.